src/java/org/apache/cassandra/guardrails/Guardrail.java

```diff
@@ -633,80 +622,116 @@ private void ensureUpToDate()
             {
                 // Setting cachedAllowed first so that on a parse failure we leave everything as it previously
                 // was (not that we'd expect that matter but ...).
-                cachedDisallowed = current.stream()
-                                          .map(parser)
-                                          .collect(Collectors.toCollection(HashSet::new));
+                cachedValues = current.stream()
+                                      .map(parser)
+                                      .collect(Collectors.toCollection(() -> new HashSet<>(current.size())));
+                                      // FIXME: Note DB-4909 used SetsFactory
+                                      // .collect(SetsFactory.collectorForSize(current.size()));
                 cachedRaw = current;
             }
             catch (Exception e)
             {
-                // This catches parsing errors. Hopefully, this shouldn't happen as guardrails settings should have
+                // This catch parsing errors. Hopefully, this shouldn't happen as guardrails settings should have
                 // been sanitized, but ...
                 // Also, we catch the exception to add a meaningful error message, but rethrow otherwise: if a
-                // guardrail has been configured, it's presumably to avoid bad things to go in, so we don't want to
+                // guardrail has been configured, it's presumably to avoid bad things to go in, so we don't wan to
                 // take the risk of letting it go if there is a misconfiguration.
                 throw new RuntimeException(format("Error parsing configured setting for guardrail on %s. This "
-                                                  + "is a bug and should not have happened."
+                                                  + "should not have happened, please contact DataStax support."
```

Comment:
Just to double-check, is it ok to reference DataStax support in the open-source version?

Reply:
Looks like I copied a bit too much from DB-4909 over that accounts for the above inconsistencies; will fix them.